crates/trigger-http/src/lib.rs

```diff
@@ -535,6 +535,16 @@ impl OutboundWasiHttpHandler for HttpRuntimeData {
             anyhow::bail!("destination-not-allowed (error 1)")
         }
 
+        let body_size_hint = request.request.body().size_hint().exact();
+        let headers = request.request.headers_mut();
+
+        if let Some(size) = body_size_hint {
+            if headers.get("content-legnth").is_none() && headers.get("transfer-encoding").is_none()
+            {
+                headers.insert("content-length", size.into());
+            }
+        }
```

Comment:
...typos like this :slightly_smiling_face: 
> `"content-legnth"`

```suggestion
        if let Some(size) = request.request.body().size_hint().exact() {
            let headers = request.request.headers_mut();
            if size > 0
                && headers.contains_key(CONTENT_LENGTH)
                && headers.contains_key(TRANSFER_ENCODING)
            {
                headers.insert(CONTENT_LENGTH, size.into());
            }
        }
```

(not sure about `size > 0` but seems right?)